### ISSUE TITLE
Error importing mysql.new.sql into Mariadb

### ISSUE THREAD

**Issue by poVoq** [processone/ejabberd#2237]
> What version of ejabberd are you using?

18.01

> What operating system (version) are you using?

Ubuntu 16.04

> How did you install ejabberd (source, package, distribution)?

.deb

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

Trying to set up SQL database use via this tutorial:
https://docs.ejabberd.im/admin/databases/mysql/
and importing the new shemata mysql.new.sql into MariaDB I get the following error:
`ERROR 1170 (42000) at line 425: BLOB/TEXT column 'server_host' used in key specification without a key length`

**Comment by kevin71020** [comment]
In my experience, compile  source code with --enable-new-sql-schema

SQL insert error when install ejabberd register user #2183

**Comment by digz6666** [comment]
Try add length to the server_host on index:
```sql
CREATE UNIQUE INDEX i_push_session_susn ON push_session (server_host(191), username(191), service(191), node(191));
```

**Comment by digz6666** [comment]
BTW why its used text data type for fields if store fixed length data there?
For some tables username is VARCHAR(191), for some tables its TEXT.
And for some indexes server_host is TEXT(191) and for some indexes its TEXT(75) etc.

**Comment by alexeyshch** [comment]
It was fixed in bef7d1ba.  JID parts can be up to 1023 bytes in length, but InnoDB supports indexes up to 767 bytes, and as it uses 4 bytes per character it results in floor(767/4)=191 limit.  No idea where 75 came from.
